centreon/www/class/centreon-clapi/centreonLDAP.class.php

```diff
@@ -333,51 +315,52 @@ public function setparam($parameters = array())
         }
         $arId = $this->getLdapId($params[0]);
         if (is_null($arId)) {
-            throw new CentreonClapiException(self::OBJECT_NOT_FOUND . ":" . $params[0]);
+            throw new CentreonClapiException(self::OBJECT_NOT_FOUND . ':' . $params[0]);
         }
-        if (in_array(strtolower($params[1]), array('name', 'description', 'enable'))) {
-            if (strtolower($params[1]) == 'name') {
-                if (!$this->isUnique($params[2], $arId)) {
+        if (in_array(mb_strtolower($params[1]), ['name', 'description', 'enable'], true)) {
+            if (mb_strtolower($params[1]) === 'name') {
+                if (! $this->isUnique($params[2], $arId)) {
                     throw new CentreonClapiException(self::NAMEALREADYINUSE . ' (' . $params[2] . ')');
                 }
             }
             $this->db->query(
-                "UPDATE auth_ressource SET ar_" . $params[1] . " = ? WHERE ar_id = ?",
-                array($params[2], $arId)
+                'UPDATE auth_ressource SET ar_' . $params[1] . ' = ? WHERE ar_id = ?',
+                [$params[2], $arId]
             );
-        } elseif (isset($this->baseParams[strtolower($params[1])])) {
-            if (strtolower($params[1]) == 'ldap_contact_tmpl') {
+        } elseif (isset($this->baseParams[mb_strtolower($params[1])])) {
+            if (mb_strtolower($params[1]) === 'ldap_contact_tmpl') {
                 if (empty($params[2])) {
                     throw new CentreonClapiException(self::MISSINGPARAMETER);
                 }
                 $contactObj = new CentreonContact($this->dependencyInjector);
                 $params[2] = $contactObj->getContactID($params[2]);
             }
-            if (strtolower($params[1]) === 'ldap_default_cg' && !empty($params[2])) {
+            if (mb_strtolower($params[1]) === 'ldap_default_cg' && ! empty($params[2])) {
                 $contactGroupObj = new CentreonContactGroup($this->dependencyInjector);
                 $params[2] = $contactGroupObj->getContactGroupID($params[2]);
             }
             $this->db->query(
-                "DELETE FROM auth_ressource_info WHERE ari_name = ? AND ar_id = ?",
-                array($params[1], $arId)
+                'DELETE FROM auth_ressource_info WHERE ari_name = ? AND ar_id = ?',
+                [$params[1], $arId]
             );
             $this->db->query(
```

Comment:
Please use a prepare + execute to avoid this legacy call to a modified query signature from PDO 🙏